docs(eos_cli_config_gen): Refactor Readme file for better readability

titom73 opened the pull request:
## Change Summary

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

N/A

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Refactor the README file to be more readable and close to documentation generated by the role itself.

## How to test

Run development stack and connect to port 8000

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

titom73 committed:
Refactor EOS CLI CONFIG GEN Readme

titom73 committed:
Remove link to GNS3 as they use browser validation

titom73 committed:
Fix link redirection for vEOS

titom73 committed:
Move TOC to the right side for better navigation

titom73 commented:
Need to change base branch to not target devel

titom73 committed:
Add missing TOC title

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -7,90 +7,95 @@
   - [Requirements](#requirements)
   - [Input Variables](#input-variables)
-    - [Terminal Settings](#terminal-settings)
+    - [Management](#management)
```
Missing the following sections:
- Terminal Settings
- Load Interval
- Queue Monitor Length
- Queue Monitor Streaming
- Domain Lookup
- Speed-Group Settings
- MAC Address-table

titom73 replied:
Good catch. Fixed with commit 5951187

carlbuchmann commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -1316,29 +1154,95 @@ peer_filters:
-### Router BGP Configuration
+### Routing
+
+### Service Routing Protocols Model
```
missing a "#" so it appears under section routing.

titom73 replied:
Good catch. Fixed with commit 5951187

carlbuchmann reviewed (changes requested):
Please see comments!

titom73 committed:
docs: Update README file as per code review

Add missing sections:

- Terminal Settings
- Load Interval
- Queue Monitor Length
- Queue Monitor Streaming
- Domain Lookup
- Speed-Group Settings
- MAC Address-table

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -137,289 +149,351 @@ Requirements are located here: [avd-requirements](../../README.md#Requirements)
+    ip_address: < IPv4_address/Mask >
+    ipv6_enable: < true | false >
+    ipv6_address: < IPv6_address/Mask >
+    gateway: <IPv4 address of gateway>
```
I cannot see this being read in any template.

titom73 replied:
Remove it from documentation with commit 8c97a1a

ClausHolbechArista commented on ansible_collections/arista/avd/roles/eos_cli_config_gen/README.md:
```diff
@@ -137,289 +149,351 @@ Requirements are located here: [avd-requirements](../../README.md#Requirements)
+    ipv6_enable: < true | false >
+    ipv6_address: < IPv6_address/Mask >
+    gateway: <IPv4 address of gateway>
+    ipv6_gateway: <IPv6 address of gateway>
```
I cannot see this being read in any template.

titom73 replied:
Remove it from documentation with commit 8c97a1a

ClausHolbechArista reviewed (changes requested):
This helps a lot. We could consider sorting each level alphabetically so it is easy to know where to add new stuff. What do you think?
I found a two knobs not implemented in the templates, which I think we should remove from readme. I have not checked all knobs throughout. I stopped at aaa-authentication since I didn't find other discrepancies.

titom73 committed:
docs(eos_cli_config_gen): Remove unused keys from management_interfaces

titom73 committed:
docs(eos_cli_config_gen): Sort all CLI section and subsection aphabetically

titom73 commented:
> This helps a lot. We could consider sorting each level alphabetically so it is easy to know where to add new stuff. What do you think?
> I found a two knobs not implemented in the templates, which I think we should remove from readme. I have not checked all knobs throughout. I stopped at aaa-authentication since I didn't find other discrepancies.

Full documentation has been sorted alphabetically per section and also under each section with commit 3f03226

ClausHolbechArista reviewed (approved):
Looks **Great** To Me

ClausHolbechArista commented:
Carl's requested changes have been done, so I will go ahead and merge
